src/libponyrt/lang/socket.c

```diff
@@ -568,7 +568,7 @@ static bool os_connect(pony_actor_t* owner, int fd, struct addrinfo *p,
     return false;
   }
 
-#ifdef PLATFORM_IS_LINUX
+#if defined(PLATFORM_IS_LINUX) || defined(PLATFORM_IS_FREEBSD) || defined(PLATFORM_IS_MACOSX)
```

Comment:
i think at this point this could be "not windows" or perhaps better, we have a define that is set per platform for "supports one shot".

I'm not completely sure on that.